tracer/src/Datadog.Trace/Telemetry/Transports/TelemetryTransportStrategy.cs

```diff
@@ -0,0 +1,64 @@
+﻿// <copyright file="TelemetryTransportStrategy.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.StreamFactories;
+using Datadog.Trace.Agent.Transports;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.HttpOverStreams;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Telemetry.Transports;
+
+internal static class TelemetryTransportStrategy
```

Comment:
nit (cause I don't know if it would be more complicated)- but basically this class is almost a copy of `TracesTransportStrategy`. Have you looked at using one class instead and passing relevant parameters? If you have and you think it doesn't make sense, fine by me.

Reply:
>Have you looked at using one class instead and passing relevant parameters

I considered it, but I don't think it actually simplifies much, as we either have to allocate parameters that may not be used, or we pass in funcs which gets a bit messy. I agree it's a candidate for refactoring though (especially after we add the .NET 6 target), and it may make sense to extract some to helper methods. I'd like to do all that in a separate PR though.

Reply:
Makes sense